searx/engines/bing.py

```diff
@@ -28,19 +28,19 @@
 """
 # pylint: disable=too-many-branches, invalid-name
 
-from typing import TYPE_CHECKING
 import base64
-import datetime
 import re
-import uuid
+import time
+from typing import TYPE_CHECKING
 from urllib.parse import parse_qs, urlencode, urlparse
-from lxml import html
+
 import babel
 import babel.languages
+from lxml import html
 
-from searx.utils import eval_xpath, extract_text, eval_xpath_list, eval_xpath_getindex
-from searx.locales import language_tag, region_tag
 from searx.enginelib.traits import EngineTraits
+from searx.locales import language_tag, region_tag
+from searx.utils import eval_xpath, eval_xpath_getindex, eval_xpath_list, extract_text
```

Comment:
@jazzzooo to keep the diff focused on the functional modifications of a patch, it is better not to reformat or reorder the code to much.  Sometimes it is OK to reformat for better readability of the code, but here by example in this diff hunk only the `import uuid` has been removed and the readability is not much better than before (BTW: import of *typings* is better placed on the top of the imports).

Reply:
@jazzzooo in commit https://github.com/return42/searxng/commit/90f06b3f1fa7c73dc4c15bfa7c3cca832033d63e I removed the hunks not needed for this patch .. if it is OK for you, I would like to reset your branch to this commit.

Reply:
> @return42 yep, go ahead

thanks / done! .. I will continue my review of this PR the next days .. could take a while, as you have might noticed bing isn't trivial :o

Reply:
Yeah no worries. If you'd like, we could merge the updated User-Agent separately, it could help other engines that are getting blocked :thinking:

Reply:
>  If you'd like, we could merge the updated User-Agent separately, it could help other engines that are getting blocked 🤔

If you have time for, it would be cool when you create a PR for the User-Agent fix :+1: :+1: